lib/core/types.hpp

```diff
@@ -40,4 +41,23 @@ using ImuData     = std::array<Float, kNumImus>;
 using EncoderData = std::array<std::uint32_t, kNumEncoders>;
 using KeyenceData = std::array<std::uint32_t, kNumKeyence>;
 
+// data produced by the accelerometer sensor
+struct acceleration_struct {
```

Comment:
Naming.

1. Upper, camel-case.
2. Don't call things struct, int, float, array, or any other thing that is obvious from the declaration.

Just `Acceleration` should do fine. Actually, in line with the others, I'd call it `RawAccelerationData`.